src/consensus/pool/slot_state.rs

```diff
@@ -167,7 +167,11 @@ impl SlotState {
             VoteKind::NotarFallback(_, _) => {
                 let block_hash = vote.block_hash().unwrap().clone();
                 let outputs = self.count_notar_fallback_stake(&block_hash, voter_stake);
-                self.votes.notar_fallback[v].push((block_hash, vote));
+                assert!(
```

Comment:
in a separate PR, I would like to combine insertion with checking for duplicates, etc.  That will be cleaner I think.